packages/core/src/application.ts

```diff
@@ -312,6 +314,23 @@ export class Application extends Context implements LifeCycleObserver {
     this.setState('started');
   }
 
+  /**
+   * Register a function to be called when the application starts.
+   *
+   * This is a shortcut for adding a binding for a LifeCycleObserver
+   * implementing a `start()` method.
+   *
+   * @param fn The function to invoke, it can be synchronous (returning `void`)
+   * or asynchronous (returning `Promise<void>`).
+   * @returns The LifeCycleObserver binding created.
+   */
+  public onStart(fn: () => ValueOrPromise<void>): Binding<LifeCycleObserver> {
+    const key = `life-cycle-observers.${fn.name || '<onStart>'}-${uuid()}`;
```

Comment:
Let's use `CoreBindings.LIFE_CYCLE_OBSERVERS` as the namespace.

Reply:
Let's use [BindingKey.generate](https://github.com/strongloop/loopback-next/blob/master/packages/context/src/binding-key.ts#L133) to generate the key.

Reply:
> Let's use `CoreBindings.LIFE_CYCLE_OBSERVERS` as the namespace.

+1, will do. Thank you for pointing out this constant, I was not aware of it.



> Let's use [BindingKey.generate](https://github.com/strongloop/loopback-next/blob/master/packages/context/src/binding-key.ts#L133) to generate the key.

I want to keep function name in the binding key. `BindingKey.generate` does not support that :(

Reply:
I briefly considered enhancing `BindingKey.generate` with a new optional argument `name`, but such change requires additional tests & docs to be written, which seems like too much overhead for what I actually need in this pull request.